libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1375,6 +1375,24 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			setLValue<MemoryItem>(_memberAccess, *_memberAccess.annotation().type);
 			break;
 		}
+		case DataLocation::CallData:
+		{
+			solUnimplementedAssert(!type.isDynamicallyEncoded(), "");
+			m_context << type.calldataOffsetOfMember(member) << Instruction::ADD;
+			// For non-value types the calldata offset is returned directly.
+			if (_memberAccess.annotation().type->isValueType())
+			{
+				solAssert(_memberAccess.annotation().type->calldataEncodedSize(false) > 0, "");
+				CompilerUtils(m_context).loadFromMemoryDynamic(*_memberAccess.annotation().type, true, false, false);
```

Comment:
```suggestion
				CompilerUtils(m_context).loadFromMemoryDynamic(*_memberAccess.annotation().type, true, true, false);
```
(struct members are always padded)

Reply:
I don't think so - this will end up in ``CompilerUtils.cpp#L1261`` and will make sure everything with ``calldataEncodedSize(false) != 32`` will be cleaned (or at some point validated I guess).

Reply:
I was in fact wrong, it will trigger the shifting in ``loadFromMemoryHelper`` - which we may in fact *not* want here. I'll check and recheck with some more tests, but I think you're right in that it should be ``true`` here.